llama_index/core/base_retriever.py

```diff
@@ -1,19 +1,32 @@
 """Base retriever."""
 from abc import abstractmethod
-from typing import List, Optional
+from typing import Any, Dict, List, Optional
 
+from llama_index.bridge.pydantic import Field
 from llama_index.callbacks.base import CallbackManager
 from llama_index.callbacks.schema import CBEventType, EventPayload
+from llama_index.core.query_pipeline.query_component import (
+    ChainableMixin,
+    InputKeys,
+    OutputKeys,
+    QueryComponent,
+    validate_and_convert_stringable,
+)
 from llama_index.prompts.mixin import PromptDictType, PromptMixin, PromptMixinType
 from llama_index.schema import NodeWithScore, QueryBundle, QueryType
 from llama_index.service_context import ServiceContext
 
 
-class BaseRetriever(PromptMixin):
+class BaseRetriever(ChainableMixin, PromptMixin):
     """Base retriever."""
 
+    # callback_manager: CallbackManager = Field(
+    #     default_factory=CallbackManager, exclude=True
+    # )
+
     def __init__(self, callback_manager: Optional[CallbackManager] = None) -> None:
         callback_manager = callback_manager or CallbackManager()
```

Comment:
oh, this was never assigned anywhere 😅 Was it causing problems to do `self.callback_manager = ...` ?

Reply:
was this changed?

Reply:
oh oops i had it in the non-fork version

Reply:
check now